[AIRFLOW-5842] Switch to Debian buster image. Depends on [AIRFLOW-5704]

potiuk opened the pull request:
NOTE FOR REVIEWER: This change depends on  #6373 . Please check only last commit.

As part of the production migration we switch to buster image
which is latest released LTS version.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5842

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

potiuk committed:
[AIRFLOW-5704] Improve Kind Kubernetes scripts for local testing

* Fixed problem that Kubernetes tests were testing latest master
  rather than what came from the local sources.
* Moved Kubernetes scripts to 'in_container' dir where they belong now
* Kubernetes tests are now better suited for running locally
* Kubernetes cluster is not deleted until environment is stopped
* Kubernetes image is build from pulled image + mounted sources
* Kubectl version name is corrected in the Dockerfile
* KUBERNETES_VERSION can now be used to select Kubernetes version
* Running kubernetes scripts is now easy in Breeze --start-kind-cluster
* Instructions on how to run Kubernetes tests are updated
* Better flags in Breeze are used to run Kubernetes environment/tests
* The old "bare" environment is replaced by --no-deps switch

potiuk committed:
[AIRFLOW-5842] Switch to debian buster image

As part of the production migration we switch to buster image
which is latest released LTS version.

mik-laj commented on Dockerfile:
```diff
@@ -263,84 +255,70 @@ ENV AIRFLOW_REPO=${AIRFLOW_REPO}
+# In case of CI builds we want to pre-install master version of airflow dependencies so that
+# We do not have to always reinstall it from the scratch and loose time for that.
+# CI build is optimised for build speed
+RUN pip install --user \
```
Should we add `--no-cache-dir` here?

mik-laj replied:
Oh. I have now seen that you set environment variables on the 55 line, but maybe you should delete the variables and make it more implicit? It also remove 1 docker layer.

feluelle commented on BREEZE.rst:
```diff
@@ -79,11 +79,11 @@ images maintained on the Docker Hub in the ``apache/airflow`` repository.
+  (size of ~1GB). Its tag follows the pattern of ``<BRANCH>-python<PYTHON_VERSION>-ci``
   (for example, ``apache/airflow:master-python3.6-ci``). The image is built using the
   `<Dockerfile>`_ Dockerfile.
+* **Production-ready image** that is used TODO
```
Can we have more information on this?

potiuk committed:
[AIRFLOW-6489] Add BATS support for Bash unit testing

We have far too much bash code around that is not automatically tested.

This is the first step to change it (simplifications and more tests are coming
soon).

feluelle reviewed (approved):
@potiuk What is the status on this?

potiuk committed:
[AIRFLOW-6491] Improve handling of Breeze parameters

While working on improving the way we run Kubernetes tests, we found out that I
need to fix handling of parameters - we change Kubernetes version used via Kind
and the old versions are no longer valid, however it was not properly
removed/saved.

We use the opportunity to add automated tests for that feature.

(cherry picked from commit 38dea9132d1fa36f4fbe871e2ab037be5ad3fab2)
